File: android/src/main/kotlin/com/example/flutter_pigeon_plugin/FlutterPigeonPlugin.kt
package com.example.flutter_pigeon_plugin

import android.util.Log
import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** FlutterPigeonPlugin */
class FlutterPigeonPlugin : FlutterPlugin, Message.FlutterMessage {
    lateinit var nativeMessage: Message.NativeMessage

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        Message.FlutterMessage.setup(flutterPluginBinding.binaryMessenger, this)
        nativeMessage = Message.NativeMessage(flutterPluginBinding.binaryMessenger)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {

    }

    override fun flutterSendMessage(msg: Message.F2NMessage) {
        Log.d("FlutterPigeonPlugin", msg.msg.toString())
        var respMsg: Message.N2FMessage = Message.N2FMessage()
        respMsg.msg2 = "Android的返回值"
        nativeMessage.nativeSendMessage(respMsg, {})

    }


}

File: lib/flutter_pigeon_plugin.dart
import 'package:flutter_pigeon_plugin/message.dart';

class FlutterPigeonPlugin extends NativeMessage {
  static final FlutterMessage _flutterMessage = FlutterMessage();

  factory FlutterPigeonPlugin() => _getInstance();

  static FlutterPigeonPlugin get instance => _getInstance();

  static FlutterPigeonPlugin? _instance = null;

  static FlutterPigeonPlugin _getInstance() {
    _instance ??= FlutterPigeonPlugin._internal();
    NativeMessage.setup(_instance);
    return _instance!;
  }

  FlutterPigeonPlugin._internal();

  void sendMessage() {
    F2NMessage message = F2NMessage();
    message.msg = "这是一条从Flutter发往Native的消息";
    _flutterMessage.flutterSendMessage(message);
  }

  @override
  void nativeSendMessage(N2FMessage msg) {
    print("收到了Native发来的消息：${msg.msg2}");
  }
}
